spark/sql/dataframe.go

```diff
@@ -108,6 +108,10 @@ type DataFrame interface {
 	ExceptAll(ctx context.Context, other DataFrame) DataFrame
 	// Explain returns the string explain plan for the current DataFrame according to the explainMode.
 	Explain(ctx context.Context, explainMode utils.ExplainMode) (string, error)
+	// FillNa replaces null values with specified value.
+	FillNa(ctx context.Context, value any, columns ...string) (DataFrame, error)
```

Comment:
in `Unpivot() / Melt()` I used `column.Convertible` for the value type and then check that these are literals. This would be more in line with the existing implementations.

Reply:
The `unpivot` is a bit different here - it allows both columns and values being expressions, while `fillna` definition works explicitly with column names and with literal values.  

I can change to match the `Unpivot`, but from usability point of view - it could be easier for people to do:

```
df.FillNa(ctx, <some_val>)
```
or
```
df.FIllNa(ctx, <some_val>, "col1", "col2")
```

and if we need to fill specific columns with specific values, then it could be:

```
df.FillNaValue(ctx, []any{1, "string"}, []string{"col1", "col2"})
```

Reply:
Although, maybe for `FillNaValues` it would be better to use this signature:

```
FillNaValues(Context, map[string]any)
```

in this case we follow the PySpark API. 

WDYT?

Reply:
The reliance on any was something I wanted to fix some time ago to have more typesafe ways to represent more complex data types that use the same underlying type.

I will try to prepare a patch that allows you to do the following:

```golang

type LiteralLike interface {
  ToLiteralExpression() *proto.Expression
}

type ConnectInt8 int8

func (c ConnectInt8) ToLiteralExpression() *proto.Expression {
  // ...
}


func (df *dataFrameImpl) FillNaValues(Context, map[string]LiteralLike) {
  // ...
}


df.FillNaValues(ctx, map[string]LiteralLike{ "colNameA" : ConnectInt8(8) })

```

Wdyt?

Reply:
For now you can finish the PR the way that you started and I'll try to have it ready.

Reply:
Yeah, it would be good.

Reply:
If you look at this PR - https://github.com/apache/spark-connect-go/pull/96/files#diff-4d2712c2c4f3da80301639ef203e41c5e97c2023475d46883064e2329b02afe4R67-R69 this is how the usage would look like.

Reply:
You can use `PrimitiveTypeLiteral` as the argument to the function.

Reply:
yep, let's merge your PR, and I'll rebase on it.

Reply:
Thank you Martin! I've updated to use your latest code.